osquery/database/tests/database_tests.cpp

```diff
@@ -114,4 +115,75 @@ TEST_F(DatabaseTests, test_ptree_upgraded_to_rj) {
   auto ulepoch = std::stoull(query_epoch);
   EXPECT_EQ(ulepoch, 1521583712U);
 }
+
+static void ptree_to_json(const pt::ptree& pt_in, const JSON& rj_in) {
+  /* Convert from ptree to string */
+  std::stringstream ss;
+  pt::write_json(ss, pt_in);
+
+  /* Convert from ptree string to rj string */
+  std::string conversion_out;
+  EXPECT_TRUE(ptreeToRapidJSON(ss.str(), conversion_out));
+
+  /* Convert rj string back to rj */
+  JSON rj_out;
+  EXPECT_TRUE(rj_out.fromString(conversion_out));
+
+  /* Compare original rj with output */
+  EXPECT_EQ(rj_in.doc(), rj_out.doc());
```

Comment:
I am not sure if RJ documents are comparable. Right now this is giving object-comparison warnings as GTest tries to do byte compares. I would recommend comparing the string outputs.

Reply:
RJ values are comparable using the operators `==` and `!=` according to the documentation, and from what I understand from the GTest documentation this should also work. I believe you're seeing the bytes because GTest doesn't know how to print the values given RJ values don't implement `<<`.

Reply:
Ok, sorry for not checking if the document types overload compatibles.